public/app/features/alerting/unified/components/rules/EditRuleGroupModal.tsx

```diff
@@ -1,18 +1,198 @@
 import { css } from '@emotion/css';
 import React, { useEffect, useMemo } from 'react';
+import { FormProvider, RegisterOptions, useForm, useFormContext } from 'react-hook-form';
 
-import { Modal, Button, Form, Field, Input, useStyles2 } from '@grafana/ui';
+import { GrafanaTheme2 } from '@grafana/data';
+import { Stack } from '@grafana/experimental';
+import { Modal, Button, Field, Input, useStyles2, Label, Badge } from '@grafana/ui';
+import { useAppNotification } from 'app/core/copy/appNotification';
 import { useCleanup } from 'app/core/hooks/useCleanup';
 import { useDispatch } from 'app/types';
 import { CombinedRuleGroup, CombinedRuleNamespace } from 'app/types/unified-alerting';
+import {
+  RulerRulesConfigDTO,
+  RulerRuleGroupDTO,
+  RulerRuleDTO,
+  RulerGrafanaRuleDTO,
+  RulerAlertingRuleDTO,
+} from 'app/types/unified-alerting-dto';
 
 import { useUnifiedAlertingSelector } from '../../hooks/useUnifiedAlertingSelector';
-import { updateLotexNamespaceAndGroupAction } from '../../state/actions';
+import { rulesInSameGroupHaveInvalidFor, updateLotexNamespaceAndGroupAction } from '../../state/actions';
 import { checkEvaluationIntervalGlobalLimit } from '../../utils/config';
 import { getRulesSourceName } from '../../utils/datasource';
 import { initialAsyncRequestState } from '../../utils/redux';
+import { parsePrometheusDuration } from '../../utils/time';
+import { DynamicTable, DynamicTableColumnProps, DynamicTableItemProps } from '../DynamicTable';
+import { InfoIcon } from '../InfoIcon';
 import { EvaluationIntervalLimitExceeded } from '../InvalidIntervalWarning';
-import { evaluateEveryValidationOptions } from '../rule-editor/GrafanaEvaluationBehavior';
+import { MIN_TIME_RANGE_STEP_S } from '../rule-editor/GrafanaEvaluationBehavior';
+
+const MINUTE = '1m';
+interface AlertInfo {
+  alertName: string;
+  forDuration: string;
+  evaluationsToFire: number;
+}
+function ForError({ message }: { message: string }) {
+  return <Badge color="orange" icon="exclamation-triangle" text={'Error'} tooltip={message} />;
+}
+
+export const getNumberEvaluationsToStartAlerting = (forDuration: string, currentEvaluation: string) => {
+  const evalNumberMs = safeParseDurationstr(currentEvaluation);
+  const forNumber = safeParseDurationstr(forDuration);
+  if (forNumber === 0 && evalNumberMs !== 0) {
+    return 1;
+  }
+  if (evalNumberMs === 0) {
+    return 0;
+  } else {
+    const evaluationsBeforeCeil = forNumber / evalNumberMs;
+    return evaluationsBeforeCeil < 1 ? 0 : Math.ceil(forNumber / evalNumberMs) + 1;
+  }
+};
+
+export const isRulerGrafanaRuleDTO = (rule: RulerRuleDTO): rule is RulerGrafanaRuleDTO => 'grafana_alert' in rule;
+export const isAlertingRuleDTO = (rule: RulerRuleDTO): rule is RulerAlertingRuleDTO => 'alert' in rule;
```

Comment:
nit: You can try to use very similar methods defined in `public/app/features/alerting/unified/utils/rules.ts`

Reply:
done!